packages/ncids-js/src/components/nci-header/utils/nci-focus-trap.ts

```diff
@@ -0,0 +1,156 @@
+ * ```
+ */
+
+export class FocusTrap {
+	/** Active Parent Component. */
+	protected element: HTMLElement;
+	/** The Element that is the contains. */
+	protected context: HTMLElement;
+	/** First focusable element */
+	protected firstFocusableElement: HTMLElement;
+	/** Last focusable element */
+	protected lastFocusableElement: HTMLElement;
+	/** List of element types to check */
+	protected focusableElements: string;
+	/** First focusable element */
+	protected focusableContent: NodeListOf<Element>;
+	/** Callback for handling the toggle functionality.  */
+	private eventListener: EventListener = (event) => this.checkTrap(event);
+	/** Map object of the component. */
+	private static _components: Map<HTMLElement, FocusTrap> = new Map<
+		HTMLElement,
+		FocusTrap
+	>();
+
+	/**
+	 * Sets component variables and initializes component.
+	 *
+	 * @param {HTMLElement} element Component being created.
+	 * @protected
+	 */
+	protected constructor(element: HTMLElement) {
+		this.element = element;
+		this.context = element;
+
+		this.focusableElements =
+			"button, [href], input, select, textarea, [tabindex]:not([tabindex='-1'])";
+		this.focusableContent =
+			this.element?.querySelectorAll(this.focusableElements) || [];
+		this.firstFocusableElement = <HTMLElement>(
+			this.element?.querySelectorAll(this.focusableElements)[0]
+		);
+		this.lastFocusableElement = <HTMLElement>(
+			this.focusableContent[this.focusableContent.length - 1]
+		);
+
+		const existingComponent = FocusTrap._components.get(this.element);
+		if (existingComponent) {
+			existingComponent.unregister();
+		}
+
+		this.initialize();
+		FocusTrap._components.set(this.element, this);
+	}
+
+	/**
+	 * Instantiates this component of the given element.
+	 *
+	 * @param {HTMLElement} element Component being created.
+	 * @public
+	 */
+	public static create(element: HTMLElement): FocusTrap {
+		if (!(element instanceof HTMLElement)) {
+			throw 'Element is not an HTMLElement';
+		}
+		return this._components.get(element) || new this(element);
+	}
+
+	/**
+	 * Resets component to a clean state.
+	 *
+	 * @public
+	 */
+	public unregister(): void {
+		FocusTrap._components.delete(this.element);
+	}
+
+	/**
+	 * Initializes the trap.
+	 *
+	 * @private
+	 */
+	private initialize(): void {
+		return;
+	}
+	/**
+	 * Toggle the FocusTrap on or Off
+	 * @param {boolean} state State of the focus tgrap
+	 * @param {context} HTMLElement the root container of the trap
+	 * @returns
+	 */
+	public toggleTrap(state: boolean, context: HTMLElement): void {
+		if (state) {
+			this.findFocusableElements(context);
+			context.addEventListener('keydown', this.eventListener, true);
+		} else {
+			context.removeEventListener('keydown', this.eventListener, true);
+		}
+	}
+
+	/**
+	 *
+	 * @param {element} context  Find all focusable elements in the
+	 * context object
+	 *
+	 * @private
+	 */
+	private findFocusableElements(element: HTMLElement) {
+		this.context = element;
```

Comment:
I don't think we need to reassign `this.context` here? It looks like it's already been assigned to the element.
we might not need the pass `element` at all if `this.context` is already scoped to the element